blocks/product-content-block/features/product-content/default.jsx

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import PropTypes from "@arc-fusion/prop-types";
+
+import { useFusionContext } from "fusion:context";
+import getProperties from "fusion:properties";
+import getTranslatedPhrases from "fusion:intl";
+
+import { Details, Icon } from "@wpmedia/arc-themes-components";
+
+const BLOCK_CLASS_NAME = "b-product-content";
+
+const contentMapping = {
+	description: "description",
+	details: "schema.productDetails",
+};
+
+const getNestedObject = (obj, path) => path.split(".").reduce((value, el) => value[el], obj);
+
+export const ProductContentDisplay = ({ children, summary, open }) => (
+	<div className={`${BLOCK_CLASS_NAME}`}>
+		<Details summary={summary} open={open} icon={<Icon name="ChevronDown" />} childrenHTML>
+			{children}
+		</Details>
+	</div>
+);
+
+const ProductContent = ({ customFields }) => {
+	const { contentType, headline, open } = customFields;
+	const { arcSite, globalContent = {} } = useFusionContext();
+	const { locale } = getProperties(arcSite);
+	const phrases = getTranslatedPhrases(locale);
+
+	if (!Object.keys(globalContent).length || !contentType) {
+		return null;
+	}
+
+	const productData = getNestedObject(globalContent, contentMapping[contentType]);
+	const visible =
+		typeof productData === "object" && Object.prototype.hasOwnProperty.call(productData, "visible")
+			? productData?.visible
+			: true;
+
+	if (!visible) {
+		return null;
+	}
+
+	const data = {
+		summary: headline || phrases.t(`product-content.${contentType}`),
+		open: !open,
```

Comment:
this logic is a bit messy. can we call the variable closed or something else that helps explain what happening here

Reply:
it's working now, but I'm not sure if it makes sense to a new dev reading this

Reply:
Updated